kahuna/public/js/util/batch-tracking.js

```diff
@@ -0,0 +1,18 @@
+// TODO MRB: invoke function lazily, does it improve UI jank?
+export function trackAll($rootScope, key, input, fn) {
```

Comment:
This could be a [factory](https://docs.angularjs.org/guide/providers#factory-recipe) to avoid passing `$rootScope` around.